9.extension/cmd/daprd/main.go

```diff
@@ -0,0 +1,367 @@
+			}),
+			bindings_loader.NewInput("kubernetes", func() bindings.InputBinding {
+				return kubernetes.NewKubernetes(logContrib)
+			}),
+		),
+		runtime.WithOutputBindings(
+			bindings_loader.NewOutput("aws.sqs", func() bindings.OutputBinding {
+				return sqs.NewAWSSQS(logContrib)
+			}),
+			bindings_loader.NewOutput("aws.sns", func() bindings.OutputBinding {
+				return sns.NewAWSSNS(logContrib)
+			}),
+			bindings_loader.NewOutput("aws.kinesis", func() bindings.OutputBinding {
+				return kinesis.NewAWSKinesis(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.eventhubs", func() bindings.OutputBinding {
+				return eventhubs.NewAzureEventHubs(logContrib)
+			}),
+			bindings_loader.NewOutput("aws.dynamodb", func() bindings.OutputBinding {
+				return dynamodb.NewDynamoDB(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.cosmosdb", func() bindings.OutputBinding {
+				return bindings_cosmosdb.NewCosmosDB(logContrib)
+			}),
+			bindings_loader.NewOutput("gcp.bucket", func() bindings.OutputBinding {
+				return bucket.NewGCPStorage(logContrib)
+			}),
+			bindings_loader.NewOutput("http", func() bindings.OutputBinding {
+				return http.NewHTTP(logContrib)
+			}),
+			bindings_loader.NewOutput("kafka", func() bindings.OutputBinding {
+				return kafka.NewKafka(logContrib)
+			}),
+			bindings_loader.NewOutput("mqtt", func() bindings.OutputBinding {
+				return mqtt.NewMQTT(logContrib)
+			}),
+			bindings_loader.NewOutput("rabbitmq", func() bindings.OutputBinding {
+				return bindings_rabbitmq.NewRabbitMQ(logContrib)
+			}),
+			bindings_loader.NewOutput("redis", func() bindings.OutputBinding {
+				return redis.NewRedis(logContrib)
+			}),
+			bindings_loader.NewOutput("aws.s3", func() bindings.OutputBinding {
+				return s3.NewAWSS3(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.blobstorage", func() bindings.OutputBinding {
+				return blobstorage.NewAzureBlobStorage(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.servicebusqueues", func() bindings.OutputBinding {
+				return servicebusqueues.NewAzureServiceBusQueues(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.storagequeues", func() bindings.OutputBinding {
+				return storagequeues.NewAzureStorageQueues(logContrib)
+			}),
+			bindings_loader.NewOutput("gcp.pubsub", func() bindings.OutputBinding {
+				return pubsub.NewGCPPubSub(logContrib)
+			}),
+			bindings_loader.NewOutput("azure.signalr", func() bindings.OutputBinding {
+				return signalr.NewSignalR(logContrib)
+			}),
+			bindings_loader.NewOutput("twilio.sms", func() bindings.OutputBinding {
+				return sms.NewSMS(logContrib)
+			}),
+			bindings_loader.NewOutput("twilio.sendgrid", func() bindings.OutputBinding {
+				return sendgrid.NewSendGrid(logContrib)
+			}),
+		),
+
+		// You can extend Dapr by creating your own Daprd binary by doing following
+		// 1. Create a GO project and copy this "main.go". Use Dapr and Component-contrib as library in the project
```

Comment:
Proposal:

You can extend Dapr by creating your own daprd binary.  Here are the steps:
1. Create a new Go project and copy this "main.go". Use Dapr and component-contrib as libraries.
2. Uncomment the code below to add your custom components.  These are exposed via gRPC endpoints.